File: baekjoon/samsung/14888.py
from itertools import permutations

def operate(a, b, op):
    if op == 0:
        return a+b
    elif op == 1:
        return a-b
    elif op == 2:
        return a*b
    else:
        if (a < 0 and b > 0) or (a > 0 and b < 0):
            return -(-a//b)
        return a//b

n = int(input())
a = list(map(int, input().split()))
ops = list(map(int, input().split()))
op = []
for i in range(4):
    for j in range(ops[i]):
        op.append(i)
ops = list(permutations(op, n-1))
max_ = -1000000000
min_ = 1000000000
for o in ops:
    result = a[0]
    for i in range(n-1):
        result = operate(result, a[i+1], o[i])
    if result > max_:
        max_ = result
    if result < min_:
        min_ = result
print(max_)
print(min_)

File: baekjoon/samsung/14888.cc
#include <cstdio>
#include <cstdlib>

using namespace std;

int n;

int operate(int *numbers, int *ops) {
    int result = 0;
    for (int i = 1; i < n; i++) {
        switch (ops[i]) {
        case 0:
            result += numbers[i-1] + numbers[i];
            break;
        case 1:
            result += numbers[i-1] - numbers[i];
            break;
        case 2:
            result += numbers[i-1] * numbers[i];
            break;
        case 3:
            result += numbers[i-1] / numbers[i];
            break;
        }
    }
    return result;
}

int main() {
    freopen("input_14888.txt", "r", stdin);
    int i;
    int op[4] = {0,};
    scanf("%d", &n);
    int *a = (int*) malloc(sizeof(int) * n);
    for (i = 0; i < n; i++)
        scanf(" %d", a + i);
    for (i = 0; i < 4; i++)
        scanf(" %d", op + i);



    // for (i = 0; i < n; i++)
        // printf("%d ", *(a + i));
    // for (i = 0; i < 4; i++)
        // printf("%d ", *(op + i));

    free(a);
    return 0;
}
